x-pack/plugins/security/server/routes/api_keys/has.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RouteDefinitionParams } from '..';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+/**
+ * Response of Kibana Has API keys endpoint.
+ */
+export interface HasAPIKeysResult {
+  hasApiKeys: boolean;
+}
+
+export function defineHasApiKeysRoutes({
+  router,
+  getAuthenticationService,
+}: RouteDefinitionParams) {
+  router.get(
+    {
+      path: '/internal/security/has_api_keys',
+      validate: false,
+      options: {
+        access: 'internal',
+      },
+    },
+    createLicensedRouteHandler(async (context, _request, response) => {
+      try {
+        // copied logic from get.ts
+        const esClient = (await context.core).elasticsearch.client;
+        const authenticationService = getAuthenticationService();
+
+        const areApiKeysEnabled = await authenticationService.apiKeys.areAPIKeysEnabled();
+
+        if (!areApiKeysEnabled) {
+          return response.notFound({
+            body: {
+              message:
+                "API keys are disabled in Elasticsearch. To use API keys enable 'xpack.security.authc.api_key.enabled' setting.",
+            },
+          });
+        }
+
+        const apiResponse = await esClient.asCurrentUser.security.getApiKey({
+          owner: true,
+        });
+
+        const validKeys = apiResponse.api_keys.filter(({ invalidated }) => !invalidated);
```

Comment:
__question:__ you filter out invalidated keys, but not expired ones - is it intentional? If not, can we use `active_only` request parameter to avoid this additional filtering?

Reply:
@azasypkin I see your point, but I don't see the `active_only` parameter supported in the `SecurityGetApiKeyRequest` typings.

Reply:
Hmm, it should be available since 8.10.0 - https://github.com/elastic/elasticsearch/pull/98259 - let me figure this out. In the worst case I'd prefer to do something like this to filter keys on ES side instead:
```ts
const apiResponse = await esClient.asCurrentUser.security.getApiKey({
  owner: true,
  // @ts-expect-error @elastic/elasticsearch SecurityGetApiKeyRequest.active_only: boolean | undefined
  active_only: true,
});

return response.ok<HasAPIKeysResult>({
  body: {
    hasApiKeys: apiResponse.api_keys.length > 0,
  },
});
```